python/az/aro/azext_aro/custom.py

```diff
@@ -75,8 +82,13 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
     rp_client_sp = aad.get_service_principal(rp_client_id)
 
     for sp_id in [client_sp.object_id, rp_client_sp.object_id]:
-        assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
-        assign_contributor_to_routetable(cmd.cli_ctx, [master_subnet, worker_subnet], sp_id)
+        for resource in sorted(resources):
+            try:
+                if not has_network_contributor_on_resource(cmd.cli_ctx, resource, sp_id):
+                    assign_network_contributor_to_resource(cmd.cli_ctx, resource, sp_id)
+
+            except CloudError as e:
+                logger.info(e.message)
```

Comment:
we are changing the create behaviour here.  Before failures in the check or assign were fatal.  Now neither are.  I don't think that makes sense.  For now, suggest keeping the current logic to avoid adding more behaviour changes in this PR.